Packet++/header/Sll2Layer.h

```diff
@@ -0,0 +1,116 @@
+#ifndef PACKETPP_SLL2_LAYER
+#define PACKETPP_SLL2_LAYER
+
+#include "MacAddress.h"
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+	/**
+	 * @struct sll2_header
+	 * Represents SLL2 header
+	 */
+#pragma pack(push, 1)
+	struct sll2_header
+	{
+		/** Contains an Ethernet protocol type of the next layer */
+		uint16_t protocol_type;
+		/** The "Reserved (MBZ)" field is reserved, and must be set to zero */
+		uint16_t reserved_type;
+		/** The interface index field is a signed integer in network byte 
+		 * order and contains the 1-based index of the interface on which the packet was observed 
+		 **/
+		int32_t interface_index_type;
+		/** Contains a Linux ARPHRD_ value for the link-layer device type */
+		uint16_t ARPHRD_type;
+		/** Specifies whether packet was: specifically sent to us by somebody else (value=0);
+		 *  broadcast by somebody else (value=1); multicast, but not broadcast, by somebody else (value=2);
+		 *  sent to somebody else by somebody else (value=3); sent by us (value=4)
+		 **/
+		uint8_t packet_type;
+		/** Contains the length of the link-layer address of the sender of the packet. That length could be zero */
+		uint8_t link_layer_addr_len;
+		/** Contains the link-layer address of the sender of the packet; the number of bytes of that field that are
+		 *  meaningful is specified by the link-layer address length field
+		 **/
+		uint8_t link_layer_addr[8];
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class Sll2Layer
+	 * Represents an SLL2 (Linux cooked capture) protocol layer
+	 */
+	class Sll2Layer : public Layer
+	{
+	public:
```

Comment:
I think it'd be nice to add getters and setters to all relevant attributes of `sll2_header`, for example:
```cpp
uint16_t getProtocolType();
void setProtocolType(uint16_t protocolType);

uint32_t getInterfaceIndex();
void setInterfaceIndex(uint32_t interfaceIndex);

...
```

Reply:
done